### DISCUSSION TITLE
暂不支持 Ranger 中的列权限、行权限以及 Data Mask 功能

### DISCUSSION THREAD

**Discussion by liujie-yt** [apache/doris#34295] [A - General / Q&A] [answered]
Hi，

我正在尝试使用 ranger 管理 doris 权限，看到文档中提到 “暂不支持 Ranger 中的列权限、行权限以及 Data Mask 功能”
想了解一下是为什么，我看文档上提供了插件的 jar 包，如果我自己定义 ranger-servicedef-doris.json 并且开发插件 有可能实现吗？

谢谢！

**Comment by yagagagaga** [comment]
Doris 本身就有[行权限](https://doris.apache.org/zh-CN/docs/dev/admin-manual/privilege-ldap/user-privilege#%E8%A1%8C%E7%BA%A7%E6%9D%83%E9%99%90)和[列权限](https://github.com/apache/doris/pull/32659)功能。

**Reply by liujie-yt** [reply]
嗯，我们规划是在 ranger 中统一管理权限，所以想集成到一起
您觉得可行吗？

**Comment by liujie-yt** [comment]
https://doris.apache.org/zh-CN/docs/dev/admin-manual/privilege-ldap/ranger

文档上提供了ranger 插件下载，不过我没找到对应的源码，ranger 那边也没找到，请问有谁知道吗，谢谢

**Reply by yagagagaga** [reply]
https://github.com/apache/doris/tree/master/fe/fe-core/src/main/java/org/apache/doris/catalog/authorizer/ranger

**Comment by liujie-yt** [comment, marked as answer]
https://doris.apache.org/zh-CN/docs/admin-manual/auth/ranger

文档显示 2.1.3 版本会支持，已经合并啦
*★,°*:.☆(￣▽￣)/$:*.°★* 。

https://github.com/apache/doris/pull/34392
https://github.com/apache/doris/pull/32137

**Comment by gagan211** [comment]
@liujie-yt 
你好 liujie-yt，

希望你一切顺利。我想和你联系，因为我有一些关于 Ranger 与 Doris 集成的问题，想和你讨论一些细节并向你请教一些经验。

期待你的回复！

gagan

**Reply by liujie-yt** [reply]
你好，不用客气，可以直接回复你的问题，我们已经实现了 Ranger 和 Doris 的集成，虽然不是我主力完成的，不过我知道的还是会尽量回答
